File: aoc2015/02.rb
p File.open('input/02.txt')
  .each_line.map { |line| line.split('x').map(&:to_i) }
  .map { |l,w,h| a,b,c = *l*w,  *w*h,  *h*l; 2*a + 2*b + 2*c + [a,b,c].min }
  .reduce(:+)

File: aoc2015/02.hs
import Data.List.Split

area [l, w, h] = let a = l * w
                     b = w * h
                     c = h * l
                 in 2*a + 2*b + 2*c + minimum [a, b, c]

main = do
    input <- readFile "input/02.txt"
    let packages  = [ [ read x :: Int | x <- splitOn "x" p ] | p <- lines input ]
    let dimensions = map area packages
    print $ sum dimensions
